PATTERNS.md

```diff
@@ -66,6 +66,40 @@ are two Exists Patterns that would match the Events above:
 If a Field in a Pattern contains an Exists Pattern, it
 **MUST NOT** contain any other values.
 
+Exists Patterns currently only work on leaf nodes. That is to 
+say, given this event:
+
+```json
+{ "a": { "b":  1 } }
+```
+
+The following pattern will not match:
+
+```json
+{ "a": [ {"exists": true} ] }
+```
+
+We may be able to change this in future.
+
+The case of empty arrays is interesting. Consider this event:
+
+```json
+{ "a": [] }`
```

Comment:
Is the backtick after `}` supposed to be in here?

Reply:
Oops good catch.